### DISCUSSION TITLE
How to pass config to language server to ask it to format a file with less spaces

### DISCUSSION THREAD

**Discussion by m3dwards** [emacs-lsp/lsp-mode#3397] [Q&A] [answered]
I'm using doom and when I press `SPC c f` to format the current buffer, if it's a tsx file I'm getting 4 spaces when I need 2 to match other colleagues.

Looking at the messages, I'm pretty sure this is lsp-mode and the language server that's doing this. My setup appears to be `ts-ls` which I believe is using https://github.com/typescript-language-server/typescript-language-server.

Is there anyway I can influence this or pass config through lsp-mode to this language server?

Many thanks

**Comment by yyoncho** [comment, marked as answer]
Check  the value of  lsp--formatting-indent-alist for the major mode that you are using (most likely typescript-mode so you should use  typescript-indent-level to configure it).

**Reply by m3dwards** [reply]
Thank you, this tip helped me ultimately find the variables involved and that my editorconfig package wasn't updating the indent width for tsx mode. The change I ended up adding which is not related to lsp-mode but I'll put here anyway was:

```
(after! editorconfig
  (add-to-list 'editorconfig-indentation-alist '(typescript-tsx-mode typescript-indent-level)))
  ```
  
  Many thanks for setting me off on the right direction!
